inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -242,68 +242,158 @@ else if (annotation instanceof Sentence) {
                                         annotation.getEnd());
                                 mtasAnnotationFeature.addPositionRange(beginToken, endToken);
                                 tokenCollection.add(mtasAnnotationFeature);
+
+                                // Returns KB IRI label after checking if the  
+                                // feature type is associated with KB and feature value is not null
+
+                                String labelStr = "";
+                                if (feature.getType().contains(IndexingConstants.KB) && featureValue != "null") {
+                                    labelStr = getUILabel(featureValue);
+                                }
+
+                                
+                                
+                                if (!labelStr.isEmpty()) {
+
+                                    String kbType = labelStr.split(MtasToken.DELIMITER)[0];
+                                    String kbValue = labelStr.split(MtasToken.DELIMITER)[1];
+                                    
+                                    if (kbType.equals(IndexingConstants.kbConcept)) {
+                                        kbType = IndexingConstants.indexKBConcept;
+                                                
+                                    }
+                                    else if (kbType.equals(IndexingConstants.kbInstance)) {
+                                        kbType = IndexingConstants.indexKBInstance;
+                                    }
+                                    
+                                    // Index IRI feature value with their labels along with 
+                                    // annotation.feature name  
+                                    String indexedStr = annotationUiName.replace(" ", "_") + "."
```

Comment:
Create method which converts a "ui name" to a "indexed name" and call it where necessary instead of explicitly calling the replace everywhere.

Reply:
Done